tests/Operation/InsertOneTest.php

```diff
@@ -25,8 +27,16 @@ public function provideInvalidConstructorOptions()
     {
         return $this->createOptionDataProvider([
             'bypassDocumentValidation' => $this->getInvalidBooleanValues(),
+            'codec' => $this->getInvalidDocumentCodecValues(),
             'session' => $this->getInvalidSessionValues(),
             'writeConcern' => $this->getInvalidWriteConcernValues(),
         ]);
     }
+
+    public function testCodecRejectsInvalidDocuments(): void
+    {
+        $this->expectExceptionObject(UnsupportedValueException::invalidEncodableValue([]));
+
+        new InsertOne($this->getDatabaseName(), $this->getCollectionName(), [], ['codec' => new TestDocumentCodec()]);
```

Comment:
I realize `[]` is perfectly valid (would result in a document with a generated `_id` field only), I think this makes the test more readable.

```suggestion
        new InsertOne($this->getDatabaseName(), $this->getCollectionName(), ['x' => 1], ['codec' => new TestDocumentCodec()]);
```

Edit: looks like you were already using `['x' => 1]` in BulkWriteTest.